src/robot/parsing/model/visitor.py

```diff
@@ -49,19 +83,53 @@ def visit_Statement(self, node):
             ...
     """
 
-    def visit(self, node: Node):
-        visitor = self._find_visitor(type(node)) or self.generic_visit
-        visitor(node)
+    def visit(self, node: Node) -> None:
+        visitor = self._find_visitor_cached(type(node)) or self.__class__.generic_visit
+        visitor(self, node)
```

Comment:
1. You used `_find_visitor_cached` and `__find_visitor` as method names. Possibly `_find_visitor` and `_find_fisitor_from_class` could be a bit better. Either way, names should use `_` or `__` prefix consistently. Probably `_` is better.
2. `_find_visitor` could possibly return `cls.generic_visit` instead of `None` to avoid `self.__class__.generic_visit` here.
3. If the above change doesn't make sense, `type(self).generic_visit` would be a bit nicer than `self.__class__.generic_visit`.